source/dofs/dof_tools.cc

```diff
@@ -765,6 +765,127 @@ namespace DoFTools
 
 
 
+  template <typename DoFHandlerType>
+  IndexSet
+  extract_dofs_with_support_on(const DoFHandlerType &dof_handler,
+                               const std_cxx11::function< bool(const typename DoFHandlerType::active_cell_iterator &)> &predicate,
+                               const ConstraintMatrix &cm)
+  {
+    const auto predicate_local
+    = [=] (const typename DoFHandlerType::active_cell_iterator & cell) -> bool
+    {
+      return predicate(cell) && cell->is_locally_owned();
```

Comment:
Is there a reason you check ``predicate`` first? I would check ``is_locally_owned`` first (potentially faster and avoids the problem of the predicate being called for artificial cells)